src/SocketUtility.cpp

```diff
@@ -3,3 +3,69 @@
 // found in the LICENSE file.
 
 #include "SocketUtility.h"
+
+SocketUtility::WSMessage parseMessage(const std::string& message) {
+        return SocketUtility::WSMessage{};
+}
+
+std::string SocketUtility::generateAcceptKey(const std::string& seckey) {
+	return StringUtility::encodeBase64(
+	StringUtility::sha1(seckey + "258EAFA5-E914-47DA-95CA-C5AB0DC85B11")
+	);
+}
+
+SocketUtility::Frame SocketUtility::DecodeMessage(const std::string & buffer){
+	SocketUtility::Frame wsFrame;
+	int pos = 0;
+	wsFrame.fin = (uint8_t)((buffer[pos] >> 7) & 0x1);
```

Comment:
Please check buffer size before accessing it to avoid buffer index out of bound.